game/gui/components/health_bar.py

```diff
@@ -24,7 +25,7 @@ def __init__(self):
         ]
 
         self.hearts_list = arcade.SpriteList()
-        self.health = 2
+        self.health = ScoreManager.health
```

Comment:
This won't be update during the code, so there is no use in storing `self.health`.
Because this is a simple type (integer) this variable will store only a value, and will not update when `ScoreManager.health` will change.
You should just use `ScoreManager.health` in `update_hearts` instead of `self.health`

Reply:
Also, move `update_hearts` code to `update` and remove `update_hearts`

Reply:
Also, you need to run update few more things to make it work:
- in `game.py` add `self.gui.update(delta_time)` in `update` function
- in `gui.py` modify `update` function:
    - add arguments: `self` and `delta_time`
    - run `self.health_bar.update` there